src/modules/campaign/sagas.ts

```diff
@@ -27,62 +25,73 @@ export function* campaignSagas(
     space: string
     environment: string
     id: string
-    token: string
   }
 ) {
   yield takeEvery(FETCH_CAMPAIGN_REQUEST, handleFetchCampaignRequest)
 
   function* handleFetchCampaignRequest(_: FetchCampaignRequestAction) {
     try {
-      const { items, includes } = (yield call(
-        [client, 'fetchEntry'],
+      const { fields } = (yield call(
+        [client, 'fetchEntryAllLocales'],
         config.space,
         config.environment,
-        config.id,
-        ADMIN_CONTENT_TYPE,
-        config.token
-      )) as ContentfulResponse<MarketingAdminFields>
+        config.id
+      )) as ContentfulEntry<MarketingAdminFields>
 
-      if (!items || (items && items.length === 0)) {
+      if (!fields) {
         throw new Error('Failed to fetch campaign data')
       }
+      const entries = (yield call(
+        [client, 'fetchEntriesFromEntryFields'],
+        config.space,
+        config.environment,
+        fields
+      )) as Record<string, ContentfulEntry<LocalizedFields>>
 
-      const assets = (includes.Asset ?? []).reduce((acc, asset) => {
-        acc[asset.sys.id] = asset
-        return acc
-      }, {} as Record<string, ContentfulAsset>)
+      const arrayOfFields = [
+        fields as LocalizedFields,
+        ...Object.values(entries).map(entry => entry.fields)
+      ]
 
-      const entries = (includes.Entry ?? []).reduce((acc, entry) => {
-        acc[entry.sys.id] = entry
-        return acc
-      }, {} as Record<string, ContentfulEntry<LocalizedFields>>)
+      const assets = (yield call(
+        [client, 'fetchAssetsFromEntryFields'],
+        config.space,
+        config.environment,
+        arrayOfFields
+      )) as Record<string, ContentfulAsset>
 
-      const banners = Object.entries(items[0].fields).reduce(
-        (acc, [key, value]) => {
-          const fieldOnLocale = value[ContentfulLocale.enUS]
-          if (isSysLink(fieldOnLocale)) {
-            const linkedEntryId = fieldOnLocale.sys.id
-            const bannerEntry = entries[linkedEntryId]
-            if (
-              bannerEntry &&
-              bannerEntry.sys.contentType.sys.id === BANNER_CONTENT_TYPE
-            ) {
-              acc[key] = {
-                ...bannerEntry.fields,
-                id: linkedEntryId
-              } as BannerFields & { id: string }
+      const banners = arrayOfFields
```

Comment:
Would you mind explaining why are we using the `arrayOfFields` here? As far as I remember, we should only go through the fields of the admin entry to get the banner entries, right?

Reply:
good catch